srcv2/components/Overlay.js

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from 'react'
+import { Modal, Icon } from '@material-ui/core'
+
+import Section from './Section'
+import Step from './Step'
+
+const Overlay = ({
+    open,
+    setOpen,
+    buttonText,
+    buttonAction,
+    goBackAction,
+    height,
+    fullScreen,
+    title,
+    description,
+    steps
+}) => {
+
+    const renderSteps = () => {
+        return steps.map((s, i) => {
+            return (
+                <Step  
+                    title={s.title}
+                    number={i + 1}
+                />
+            )
+        })
+    }
+
+    return (
+        <div className='Overlay'>
+            <Modal
+                open={open}
+                onClose={() => { setOpen(false) }}
+                className={'fixed inset-0 flex bg-black bg-opacity-50 items-center justify-center transition-opacity duration-300 z-40'}
+            >
+                <div className={'fixed inset-5 mt-20'} onClick={() => { setOpen(false) }}>
+                    <Section backgroundColor={'bg-white'} className={`flex flex-col rounded-lg border-2 !py-3 px-6 ${fullScreen ? 'h-full' : ''} ${height ? height : ''}`}>
+                        <div className='y-3'>
+                            <Icon 
+                                className={`icon fas fa-arrow-left my-3 !w-8 ${goBackAction ? 'visible' : 'invisible'}`}
+                                onClick={() => { goBackAction() }}
+                            />
+                            <Icon 
+                                className={`icon fas fa-x !w-8 my-3 absolute top-4 right-4`}
+                                onClick={() => { setOpen(false) }}
+                            />
+                        </div>
+                        <div className='flex-grow mt-5'>
+                            <p className='text-3xl font-bold mb-2'>
+                                {title}
+                            </p>
+                            <p>
+                                {description}
+                            </p>
+                            {steps && renderSteps()}
+                        </div>
```

Comment:
All this can be replaced with a child component.

The ideal use of this component should be :
```js
<Overlay 
    open,
    setOpen,
    buttonText,
    buttonAction,
    goBackAction,
    height,
    fullScreen
>
   <div className='flex-grow mt-5'>
      <p className='text-3xl font-bold mb-2'>
         Title
      </p>
       <p>
         Description
      </p>
         steps
   </div>
</Overlay>
```

Reply:
I see, going to try it out.